Changing typescript peer dep to remove 4.x is a breaking change

## Which package?

react-intl & @formatjs/intl

## Describe the bug

I have a package using react-intl@6 and typescript@4.  Those were resolved to react-intl@6.4.4 and typescript@4.9.5 in the package-lock.json.

I updated to react-intl@6.5.5 and got the peer dep error shown below.  The only way to resolve the error properly is to do a major version update on typescript. I didn't expect this for a minor version bump of react-intl, and would consider this a breaking change by semver standard. See: https://github.com/semver/semver/issues/502

This is the commit that changed the peer dep.  I don't see anything in there that would require removing 4.x:
https://github.com/formatjs/formatjs/commit/141dca6dcc7a21ce87010b5f69e5080d74e16b9b

## To Reproduce

### Reproducible Steps/Repo

Steps to reproduce the behavior:

1. Create a temp directory and switch into it.
2. Run `npm init -y`
3. Run `npm install react-intl@6.4.4`
4. Run `npm install typescript@4`
5. Run `npm install react-intl@latest`
6. Peer dep error:

```
npm ERR! While resolving: react-intl@6.5.5
npm ERR! Found: typescript@4.9.5
npm ERR! node_modules/typescript
npm ERR!   peerOptional typescript@"5" from @formatjs/intl@2.9.9
npm ERR!   node_modules/react-intl/node_modules/@formatjs/intl
npm ERR!     @formatjs/intl@"2.9.9" from react-intl@6.5.5
npm ERR!     node_modules/react-intl
npm ERR!       react-intl@"6.5.5" from the root project
```

## Expected behavior

Peer dep major version bumps should not be done in minor version updates since they may break existing installs.

--- a/packages/intl/package.json
+++ b/packages/intl/package.json
@@ -42,7 +42,7 @@
     "@formatjs/intl-numberformat": "workspace:*"
   },
   "peerDependencies": {
-    "typescript": "5"
+    "typescript": "^4.7 || 5"
   },
   "peerDependenciesMeta": {
     "typescript": {

--- a/packages/react-intl/package.json
+++ b/packages/react-intl/package.json
@@ -145,7 +145,7 @@
   },
   "peerDependencies": {
     "react": "^16.6.0 || 17 || 18",
-    "typescript": "5"
+    "typescript": "^4.7 || 5"
   },
   "peerDependenciesMeta": {
     "typescript": {

